dataland-frontend/src/utils/DataTableDisplay.ts

```diff
@@ -43,3 +43,83 @@ export type ReportingPeriodOfDataSetWithId = {
   dataId: string;
   reportingPeriod: string;
 };
+
+/**
+ * Adds click event listeners on DataTable row headers to expand and collapse row
+ * @param expandedRowsOnClick function passes the latest list of expanded row id's
+ * @param newExpandedRowsCallback function that returns the updated (after click) list of expanded row id's
+ * @returns the map of rows and their click handlers needed for unmounting
+ */
+export function mountRowHeaderClickEventListeners(
+  expandedRowsOnClick: () => string[],
+  newExpandedRowsCallback: (newExpandedRows: string[]) => void,
+): Map<Element, EventListener> {
+  const handlerMap: Map<Element | HTMLButtonElement, EventListener> = new Map();
+  let expandedRowGroups: string[] = [];
+
+  const t = setTimeout(() => {
+    const buttonAttributeName = "data-parent-id";
+    const rows = Array.from(document.querySelectorAll("[data-row-header-click]")).filter((el) => el);
+    const rowButtons = rows
+      .map((el: Element) => {
+        const button = el.parentNode?.querySelector('button[data-pc-section="rowgrouptoggler"]');
+        if (button) {
+          button?.setAttribute(buttonAttributeName, el.id);
```

Comment:
The "?" should not be needed, since the code is only exectued if button exists (because of the if-condition).

Reply:
hehe that's true. Removed.

Reply:
👍